### ISSUE TITLE
Mark file public

### ISSUE THREAD

**Issue by elkhawajah** [badunk/multer-s3#11]
I don't see an option to pass ACL param when storing a file, is this missing ? or it can be done somehow ?

**Comment by badunk** [comment]
its not supported right now, do you have any thoughts on how it should look like?

**Comment by elkhawajah** [comment]
I forked the repo and did quick fix for my case https://github.com/elkhawajah/multer-s3/blob/master/index.js#L59-L65 

This fix does not take into account that ACL might not be passed in options, the correct fix I think should look like this

```
// uploadOptions would include params to be passed to S3 ManageUpload to putObject
var params = aws.util.merge(that.options.uploadOptions || {}, {
        Bucket: that.options.bucket,
        Key: filePath,
        ContentType: contentType,
        Body: (_stream || file.stream)
      });

// pass params
var upload = that.s3.upload(params)
```

I think above solution would solve the problem with handling params case by case, simply pass the options to aws sdk, if anything invalid aws will raise error.

**Comment by badunk** [comment]
could you submit a PR and I'll look it over?

Thanks!

**Comment by alekaizer** [comment]
I had the same problem today with the ACL, did a quick fix and made a PR!

Sorry for the Title though, haven't paid attention and left it *Updated index.js* should be *ACL options support*

My bad :)

**Comment by badunk** [comment]
Please checkout version `1.5.0`

**Comment by igneosaur** [comment]
Hey man, I'm not sure this implementation is working. I've got the latest (1.5.0) package in my project but passing an `ACL` param into the `multerS3` object has no effect.

```
var upload = multer({
  storage: multerS3({
    s3: s3,
    bucket: 'some-bucket',
    key: function (req, file, cb) {
      cb(null, Date.now().toString())
    },
    ACL: 'public-read'
  })
});
```

I don't see anywhere in the source code that passes through the `opts` as it is https://github.com/badunk/multer-s3/blob/master/index.js#L86

**Comment by badunk** [comment]
@igneosaur oh you're right, sorry about that.  let me look into it

**Comment by badunk** [comment]
@LinusU  do you see any problem with just passing the object through https://github.com/badunk/multer-s3/blob/master/index.js#L86 and explicitly setting the body if it doesn't already exist?

**Comment by LinusU** [comment]
Hmm, I think that we should have the ability to change the `ACL` per file so I would personally prefer it to be done in the same way that contentType and key is done. That is, if it's a string, that value will always be used, but if it's a function, it will be called with `(req, file, cb)` so that it can be customised on a per file basis.

**Comment by drora** [comment]
guys this is a serious issue.
without ACL control this nice module is almost useless.
any ETAs? 
I wouldn't want to move to a less nice module with ACL control...

**Comment by drora** [comment]
here is an easy and stupid workaround::

post(upload.single('file'), (req, res) => {
		s3.putObjectAcl({
			Bucket: s3conf.bucket,
			Key: `${req.file.key}`,
			ACL: s3conf.ACL
		});
		return res.json(req.file.location);
	})

**Comment by thebergamo** [comment]
Well, in a quick look at NPM I see this module https://www.npmjs.com/package/multer-s3-acl

**Comment by igneosaur** [comment]
Alright guys, I made an adjustment to the code for myself but thought I might as well throw it back here as a PR. Hopefully it's not too far from what you had in mind, @LinusU.

**Comment by thebergamo** [comment]
Neat PR @igneosaur ! will be very appreciate if @badunk accept that! Will be a lot helpful because I try implement this feature right now :p

**Comment by igneosaur** [comment]
Fix #27 has now been merged into version `2.1.0`, solving this issue.

**Comment by badunk** [comment]
thanks
